autocodesign/profiles.go

```diff
@@ -440,11 +458,25 @@ func checkProfileCertificates(profileCertificateIDs []string, certificateIDs []s
 	return nil
 }
 
-func checkProfileDevices(profileDeviceIDs []string, deviceIDs []string) error {
-	for _, id := range deviceIDs {
-		if !sliceutil.IsStringInSlice(id, profileDeviceIDs) {
+func validDeviceUDID(udid string) string {
+	r := regexp.MustCompile("[^a-zA-Z0-9-]")
+	return r.ReplaceAllLiteralString(udid, "")
+}
+
+func normalizeDeviceUDID(udid string) string {
```

Comment:
Why is this needed? Do you have an example UDID? We might add a comment with the original UDID and the transformed and why the transformation is needed.